crmsh/utils.py

```diff
@@ -2287,4 +2249,109 @@ def interface_choice():
+        if rc != 0:
+            raise ValueError(err)
+
+        # format on each line will like:
+        # 2: enp1s0    inet 192.168.122.241/24 brd 192.168.122.255 scope global enp1s0\       valid_lft forever preferred_lft forever
+        for line in out.splitlines():
+            _, nic, _, ip_with_mask, *_ = line.split()
+            if nic == "lo":
+                continue
+            # one nic might configured multi IP addresses
+            if nic not in self._nic_info_dict:
+                self._nic_info_dict[nic] = []
+            self._nic_info_dict[nic].append(Interface(ip_with_mask))
+
+        if not self._nic_info_dict:
+            raise ValueError("No address configured")
+        if self.second_heartbeat and len(self._nic_info_dict) == 1:
+            raise ValueError("Cannot configure second heartbeat line, since only one address is available")
+
+    def get_default_ip_list(self):
+        _ip_list = []
+        for nic in self.specify_nic_list:
+            if nic not in self.nic_list:
+                raise ValueError("Failed to detect IP address for {}".format(nic))
+            _ip_list.append(self._nic_info_dict[nic][0].ip)
+        if self.second_heartbeat and len(self.specify_nic_list) == 1:
+            for nic in self.nic_list:
+                if nic not in self.specify_nic_list:
+                    _ip_list.append(self._nic_info_dict[nic][0].ip)
+                    break
+        return _ip_list
+
+    def get_default_nic_list_from_route(self):
+        cmd = "ip -o route show"
+        rc, out, err = get_stdout_stderr(cmd)
+        if rc != 0:
+            raise ValueError(err)
+        res = re.search(r'^default via .* dev (.*?) ', out)
+        if res:
+            return [res.group(1)]
+        else:
+            common_warn("No default route configured")
+            return [self.nic_list[0]]
+
+    def ip_in_network(self, addr):
+        for net in self.ip_with_mask_list:
+            if ipaddress.ip_address(addr) in ipaddress.ip_interface(net).network:
+                return True
+        return False
+
+    @property
+    def default_nic_list(self):
+        return self.specify_nic_list
+
+    @property
+    def nic_list(self):
+        return list(self._nic_info_dict.keys())
+
+    @property
+    def interface_list(self):
+        _interface_list = []
+        for interface in self._nic_info_dict.values():
+            _interface_list.extend(interface)
+        return _interface_list
+
+    @property
+    def ip_list(self):
+        return [interface.ip for interface in self.interface_list]
+
+    @property
+    def ip_with_mask_list(self):
+        return [interface.ip_with_mask for interface in self.interface_list]
+
+    @property
+    def network_list(self):
+        return [interface.network for interface in self.interface_list]
+
+
+class Interface(object):
+    def __init__(self, ip_with_mask):
+        self.ip, self.mask = ip_with_mask.split('/')
+
+    def get_network(self):
```

Comment:
I think we can remove this method `get_network` method, and just set `self.nework` in the `init` method. Having `property` type method calling a method is a bit strange. In fact, it looks like this value is static.

Anyway, current solution works too